life_expectancy/cleaning.py

```diff
@@ -21,93 +20,134 @@
 
 # Define variables cleaning  and filtering data
 COMPOSED_COL = 'unit,sex,age,geo\\time'
-DECOMPOSED_COLs = ['unit', 'sex', 'age','region']
+DECOMPOSED_COLS= ['unit', 'sex', 'age','region']
 
-def clean_data(region_filter: str) -> None:
+def load_data (input_file_path: Path = INPUT_FILE_PATH) -> pd.DataFrame:
     """
-    This function processes and filters a file with european life expectancy data over the years.
+    This function loads a file with european life expectancy data over the years.
 
     Parameters:
-    None
+    input_file_path (str): The path to the input file.
 
     Returns:
-    Just saves the processed data for Portugal.
+    df_raw (pd.DataFrame): The loaded data.
     """
+    df_raw = pd.DataFrame()  # define an empty DataFrame
 
     try:
         # load data
-        with open(INPUT_FILE_PATH, encoding='utf-8') as filename:
+        with open(input_file_path, encoding='utf-8') as filename:
             df_raw = pd.read_csv(filename, sep='\t', na_values=[':'])
-    except FileNotFoundError:# pragma: no cover
-        logging.error("Input file %s not found.", INPUT_FILE_PATH)
-        return
+    except PermissionError:# pragma: no cover
+        logging.error("Permission denied to read %s.", input_file_path)
+        return df_raw
+    except FileNotFoundError as error:# pragma: no cover
+        logging.error(str(error))
+        return df_raw
+    except pd.errors.ParserError as error:# pragma: no cover
+        logging.error("Error reading file %s: %s", input_file_path, error)
+        return df_raw
+    return df_raw
+
+def __convert_datatypes(dataframe: pd.DataFrame) -> pd.DataFrame:
+    """
+        This function iterates on each group of columns to convert to the determined data type
+    """
+    #convert column data types explicitly all with exception of "value"
+    data_types = {
+        'unit': 'category',
+        'sex': 'category',
+        'age': 'category',
+        'region': 'category',
+        'year': 'int'
+    }
+    try:
+        dataframe = dataframe.astype(data_types)
+    except ValueError as error:# pragma: no cover
+        logging.error("Datatype conversion error %s", error)
+    #convert value column to numeric
+    try:
+        dataframe['value'] = pd.to_numeric(dataframe['value']\
+                                                .str.extract(r'(\d+(?:\.\d+)?)',
+                                                            expand=False), errors='raise')
+    except ValueError as error:# pragma: no cover
+        logging.error("Datatype conversion error %s", error)
```

Comment:
This could be potentially dangerous: we failed to fix the values so we are returning the dataframe as is. Doesn't that means we will be saving the wrong data in our datamart?

Reply:
good point, added raise error after logging message